### PR TITLE
[MPDX-8192] Do not allow fractional admin percentages

### PR DESCRIPTION
**Pull request by canac**
## Description

Require the admin percentage to be a whole number.

[MPDX-8192](https://jira.cru.org/browse/MPDX-8192)

## Checklist:

- [x] I have given my PR a title with the format "MPDX-(JIRA#) (summary sentence max 80 chars)"
- [x] I have applied the appropriate labels. (_Add the label "On Staging" to get the branch automatically merged into staging._)
- [x] I have requested a review from another person on the project

### CODE REVIEW

File: src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx [hunk 1]
```diff
@@ -177,7 +177,7 @@ const appealFormSchema = yup.object({
     .test(
       i18n.t('Is positive?'),
       i18n.t('Must use a positive number for Initial Goal'),
-      (value) => parseFloat(value as unknown as string) >= 0,
+      (value) => typeof value === 'number' && value >= 0,
```
Comment (dr-bizz) [inline src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx:180]: When testing on https://8192-validate-percentage.d3dytjb8adxkk5.amplifyapp.com/
I was still able to enter a number with a decimal point.

**Review by dr-bizz — CHANGES_REQUESTED** [review]
The Admin Percentage looks great. Can you add the rule of whole numbers to initial goal and letter cost?

I know I didn't specify that on the Task.
Screenshot 2024-09-12 at 9 28 59 AM

File: src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx (hunk 1 above)
Comment (canac) [reply src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx:180]: Are you saying the admin percentage validation is good but there isn't validation yet on the initial goal?

**Comment by canac** [comment]
> Can you add the rule of whole numbers to initial goal and letter cost?

@dr-bizz It kind of seems weird to me that we're trying to keep users from entering fractional amounts, but the goal still probably won't probably be a whole number (e.g. with the numbers below). It feels like we should be rounding the goal amount too if we want everything to be a whole number. Let me know what you think.

Screenshot 2024-09-12 at 10 21 31 AM

Also, the goal amount is editable in Angular but not in React. That needs fixed too.

File: src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx (hunk 1 above)
Comment (dr-bizz) [reply src/components/Tool/Appeal/InitialPage/AddAppealForm/AddAppealForm.tsx:180]: yes that is correct

**Comment by dr-bizz** [comment]
@dr-bizz It kind of seems weird to me that we're trying to keep users from entering fractional amounts, but the goal still probably won't probably be a whole number (e.g. with the numbers below). It feels like we should be rounding the goal amount too if we want everything to be a whole number. Let me know what you think.


This is how the old MPDX was, not saying it's correct. But you can bring this up with Scott and see if he wants to make the change. I'm okay with it if he is.

**Comment by dr-bizz** [comment]
I've created a task for the Goal input -  [MPDX-8208](https://jira.cru.org/browse/MPDX-8208)

**Comment by canac** [comment]
If we're duplicating Angular, the Angular version allows you to enter fractional amounts for all four fields, including the admin percentage.

Screenshot 2024-09-13 at 9 04 26 AM

**Comment by dr-bizz** [comment]
> If we're duplicating Angular, the Angular version allows you to enter fractional amounts for all four fields, including the admin percentage.
> 
> Screenshot 2024-09-13 at 9 04 26 AM

Have you clicked submit?

**Comment by canac** [comment]
@dr-bizz My bad, you're right 🤦‍♂️. I'll add the whole number validation to the three fields.

**Comment by dr-bizz** [comment]
All good. That got me the first time as well.

**Pull request merged**